packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -426,22 +426,6 @@ class AndroidSdk {
       return fileSystem.path.join(javaHomeEnv, 'bin', 'java');
     }
```

Comment:
It'd be good to add a comment at the top of this function saying it needs to be kept in sync with how `android/gradle.dart` sets `JAVA_HOME` when running Gradle. Also a comment at the relevant places in that file (the two places that https://github.com/flutter/flutter/pull/124152/files#diff-3c44d2e8279ea8d917b4fb942252d4649689154c36cf74227a1a2a879aaf1915 changes to use `_javaSdkHome`) saying they need to be kept in sync with what `findJavaBinary` does.

Reply:
@andrewkolos I agree with @gnprice here.